pkg/action/scan.go

```diff
@@ -83,20 +106,34 @@ func scanSinglePath(ctx context.Context, c malcontent.Config, path string, ruleF
 	if err != nil {
 		return nil, err
 	}
-
 	size := fi.Size()
-	fc := make([]byte, size)
 
+	fc := filePool.Get(size)
+	defer filePool.Put(fc)
 	if _, err := io.ReadFull(f, fc); err != nil {
 		return nil, err
 	}
 
-	mrs, err := yrs.Scan(fc)
+	// Immediately remove archive files read into memory
+	if isArchive {
+		os.RemoveAll(path)
+	}
```

Comment:
Should this be in a defer so we can guarantee this happens even if we encounter an err somewhere above?

Reply:
Added in [`cf0b6ca` (#867)](https://github.com/chainguard-dev/malcontent/pull/867/commits/cf0b6ca496d50982cf0d09b765fab4ef53437168).